layers/gpuav/instrumentation/gpuav_shader_instrumentor.cpp

```diff
@@ -1030,30 +1047,32 @@ bool GpuShaderInstrumentor::PreCallRecordPipelineCreationShaderInstrumentationGP
                     // not double-free the memory on us. If making any changes, we have to consider a case where the user inlines
                     // the fragment shader, but use a normal VkShaderModule in the vertex shader.
                     modified_shader_module_ci->SetCode(instrumented_spirv);
-                } else {
-                    assert(false);
-                    return false;
                 }
+            } else {
+                assert(false);
+                return false;
             }
         }
 
-        // Create instrumented pipeline library
-        // ---
-        VkPipeline instrumented_pipeline_lib = VK_NULL_HANDLE;
-        const VkResult result = DispatchCreateGraphicsPipelines(device, VK_NULL_HANDLE, 1, modified_pipeline_ci.ptr(), pAllocator,
-                                                                &instrumented_pipeline_lib);
-        if (result != VK_SUCCESS || instrumented_pipeline_lib == VK_NULL_HANDLE) {
-            InternalError(device, loc, "Failed to create instrumentde pipeline library.");
-            return false;
-        }
+        // Create instrumented pipeline library if we have instrumented one of the libraries inside of it
+        if (need_new_pipeline) {
+            VkPipeline instrumented_pipeline_lib = VK_NULL_HANDLE;
+            const VkResult result = DispatchCreateGraphicsPipelines(device, VK_NULL_HANDLE, 1, modified_pipeline_ci.ptr(),
+                                                                    pAllocator, &instrumented_pipeline_lib);
+            if (result != VK_SUCCESS || instrumented_pipeline_lib == VK_NULL_HANDLE) {
+                InternalError(device, loc, "Failed to create instrumentde pipeline library.");
```

Comment:
```suggestion
                InternalError(device, loc, "Failed to recreate instrumented pipeline library.");
```
(not sure about `recreate`?)